presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -1279,6 +1281,54 @@ public static long widthBucket(@SqlType(StandardTypes.DOUBLE) double operand, @S
         return lower;
     }
 
+    @Description("Linearly interpolate a value at x given coordinates")
+    @ScalarFunction("linear_interpolate")
+    @SqlNullable
+    @SqlType(StandardTypes.DOUBLE)
+    public static Double linearInterpolate(
+            @SqlType(StandardTypes.DOUBLE) double x,
+            @SqlType("array(double)") Block xArray,
+            @SqlType("array(double)") Block yArray)
+    {
+        int xCount = xArray.getPositionCount();
+        int yCount = yArray.getPositionCount();
+        checkCondition(xCount == yCount, INVALID_FUNCTION_ARGUMENT, "Arrays must be the same length");
+        checkCondition(xCount >= 2, INVALID_FUNCTION_ARGUMENT, "Arrays must have length >= 2");
+        checkCondition(!Double.isNaN(x) && !Double.isInfinite(x), INVALID_FUNCTION_ARGUMENT, "NaNs not supported");
+
+        double[] xPrimitiveArray = new double[xCount];
+        double[] yPrimitiveArray = new double[xCount];
+        boolean yIsNull = false;
+        for (int i = 0; i < xCount; i++) {
+            Double xValue = DOUBLE.getDouble(xArray, i);
+            Double yValue = DOUBLE.getDouble(yArray, i);
+            checkCondition(!xArray.isNull(i), INVALID_FUNCTION_ARGUMENT, "x array must be strictly increasing");
```

Comment:
How big these arrays tend to be? There are way too many conditionals in the for loop. It would probably be more efficient if you take the conditional checks out. For example, do the null check before processing the data.
```
checkCondition(IntStream.range(0, xCount).noneMatch(i -> xArray.isNull(i)), "....");
```